model/search_pipeline/search_pipeline.smithy

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+//  The OpenSearch Contributors require contributions made to
+//  this file be licensed under the Apache-2.0 license or a
+//  compatible open source license.
+
+$version: "2"
+namespace OpenSearch
+
+structure SearchPipelineStructure{
+    version: Integer,
+    request_processors: RequestProcessorsList,
+    response_processors: ResponseProcessorsList
```

Comment:
There's also a concept of a `phase_results_processors`:
https://opensearch.org/docs/latest/search-plugins/search-pipelines/search-processors/#search-phase-results-processors
https://opensearch.org/docs/latest/search-plugins/search-pipelines/normalization-processor/#creating-a-search-pipeline